packages/flutter/lib/src/material/material_theme.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+// Examples can assume:
+// late BuildContext context;
+
+/// Overrides the default properties values for descendant [Material] widgets.
+///
+/// Descendant widgets obtain the current [MaterialThemeData] object
+/// using `MaterialTheme.of(context)`. Instances of [MaterialThemeData] can
+/// be customized with [MaterialThemeData.copyWith].
+///
+/// Typically a [MaterialThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.materialTheme].
+///
+/// All [MaterialThemeData] properties are `null` by default.
+/// When null, the [Material] will use the values from [ThemeData]
+/// if they exist, otherwise it will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class MaterialThemeData with Diagnosticable {
+  /// Creates the set of color, style, and size properties used to configure [Material].
+  const MaterialThemeData({
+    this.color,
+    this.shadowColor,
```

Comment:
If we are including `shadowColor` we should probably include `elevation` and `surfaceTintColor`.

Reply:
I agree that having a theme for Material is a little odd. Also: most of the Material properties implicitly animate (see https://api.flutter.dev/flutter/material/Material/animationDuration.html) which can be very confusing if the Material is part of some larger component whose properties are being animated. 

I approved this PR however oon second thought, I think we'd be better off limiting theme support for components, rather than building blocks.

Reply:
>Is this just so we can deprecate ThemeData.shadowColor?

This is in service of deprecating `ThemeData.canvasColor`, which is a default color for `Material`. 

`ThemeData.cardColor` is also used by Material with [MaterialType.card] but I'm opting to deprecate it for `CardTheme.color`. `ThemeData.shadowColor` is being deprecated for `ColorScheme.shadow`, as you said.

So question: should we leave `ThemeData.canvasColor` in place while deprecating `ThemeData.cardColor`? Leave both in place? Both are defaults for `Material`.

Reply:
Do you think `canvasColor` is that relevant/used? I didn't even know it exists. I usually use the Scaffold background color only, I never set on MaterialApp.

Reply:
Rather than creating a Material theme, it might be better to replace the reference to canvasColor with a ColorScheme color that's a good match for canvasColor when useMaterial3 is true.